test/Parse/number_identifier_errors.swift

```diff
@@ -36,23 +36,23 @@ typealias 11.0 = Int
 // expected-error@-1 {{typealias name can only start with a letter or underscore, not a number}}
 typealias 12typealias = Int
 // expected-error@-1 {{typealias name can only start with a letter or underscore, not a number}}
-// expected-error@-2 {{expected a digit in integer literal}}
+// expected-error@-2 {{'t' is not a valid digit in integer literal}}
 
 struct 13 {}
 // expected-error@-1 {{struct name can only start with a letter or underscore, not a number}}
 struct 14.0 {}
 // expected-error@-1 {{struct name can only start with a letter or underscore, not a number}}
 struct 15struct {}
 // expected-error@-1 {{struct name can only start with a letter or underscore, not a number}}
-// expected-error@-2 {{expected a digit in integer literal}}
+// expected-error@-2 {{'s' is not a valid digit in integer literal}}
 
 enum 16 {}
 // expected-error@-1 {{enum name can only start with a letter or underscore, not a number}}
 enum 17.0 {}
 // expected-error@-1 {{enum name can only start with a letter or underscore, not a number}}
 enum 18enum {}
 // expected-error@-1 {{enum name can only start with a letter or underscore, not a number}}
-// expected-error@-2 {{expected a digit in floating point exponent}}
+// expected-error@-2 {{'n' is not a valid digit in floating point exponent}}
```

Comment:
This is, I guess, covered by rdar://problem/32901394 (I'm not sure if there's a JIRA for it).